src/frequenz/sdk/microgrid/_data_pipeline.py

```diff
@@ -170,6 +171,25 @@ def ev_charger_pool(
             )
         return self._ev_charger_pools[key]
 
+    def grid(
+        self,
+    ) -> _grid.Grid | None:
+        """Return the grid instance.
+
+        If a Grid instance doesn't exist, a new one is created and returned.
+
+        Args:
+            components: The components to use when fetching the grid frequency.  If None,
+                        the component will be fetched from the registry.
+
+        Returns:
+            A Grid instance.
+        """
+        if _grid.get() is None:
+            component_graph = connection_manager.get().component_graph
+            _grid.initialize(component_graph.components())
+        return _grid.get()
```

Comment:
The idea was to remove the `_grid.initialize()` and just move the initialization to `_DataPipeline` like we do with the `battery_pool`, etc. (so this method is just implemented as `return _get().grid()`)

I mean, in terms of interface, this is enough for v1.0 (as we are getting the public interface we wanted), but in terms of making the code more maintainable I would do the move, so we have only one way to initialize *microgrid objects*. If it's not a lot of work I would do this for v1.0 but if it is we can leave it for afterwards. Also I didn't check if it is 100% feasible and if it really fits `_DataPipeline`, which is running some actors, so there is also the chance that this approach is not great either, please don't trust me blindly and judge it from yourself :)

Reply:
we call grid modules .initialize in the `_DataPipeline` initialization already. these 3 lines have been a left-over due to the fact that i needed a way to re-initializell in the unit tests. But we've agreed to another way on the test-side already. I've dropped the deadbeaf :)

Reply:
I guess you forgot to push then :brain: :dash:

Reply:
now